packages/pong-engine/src/physics.ts

```diff
@@ -53,6 +51,32 @@ const bounceArkanoid = (
   };
 };
 
+const bounceArkanoid = (
+  ball: GameBall,
+  paddle: GamePaddle,
+  deltaTime: number,
+): GameBall => {
+  const paddleBallDistance = ball.x - paddle.x;
+  const bounceAngle =
+    MAX_PADDLE_BOUNCE_ANGLE -
+    (paddleBallDistance / PADDLE_WIDTH) *
+      (MAX_PADDLE_BOUNCE_ANGLE - MIN_PADDLE_BOUNCE_ANGLE);
+  return getArkanoidBounce(ball, bounceAngle, deltaTime);
+};
+
+const bounceArkanoidOpponent = (
```

Comment:
`bounceArkanoid` and this are the same except for `MXX_PADDLE_OPPONENT_BOUNCE_ANGLE` vs `MXX_PADDLE_BOUNCE_ANGLE`. Why not packing these player dependant constants into a config object, and passing this object to different functions to return the player specific `playerA` and `playerB` behaviour?

Reply:
This has, kind of, already been done, we’ve move the common part of  both functions to a common one. These angle calculations are specific to each bounce and this code has be somewhere. Also does not make sense to move the angles outside it because is only responsibility of the bounce function to know about them.